### ISSUE TITLE
Failed to deserialize REST JSON data in `deposits exit` call

### ISSUE THREAD

**Issue by heeckhau** [status-im/nim-beacon-chain#5362]
I ran into this error while trying to exit a Goerli  validator :
```
(FAT 2023-08-25 07:21:26.306+00:00 Failed to obtain the config spec of the beacon node reason="Unable to decode REST response, error [Serialization error]")
```

With debug mode enabled, I got the attached output:
`/home/user/nimbus-eth2/build/nimbus_beacon_node deposits exit --validator=546123 --print --log-level=DEBUG`
```
Failed to deserialize REST JSON data       err="<data>(1, 2889) Non-decimal character encountered: 0x00" data="{\"data\":{\...
```
Full log:
```
DBG 2023-08-25 07:21:26.305+00:00 Sending REST request to remote server      remote=127.0.0.1:5052 request=/eth/v1/beacon/genesis http_method=GET
DBG 2023-08-25 07:21:26.305+00:00 Got REST response headers from remote server remote=127.0.0.1:5052 request=/eth/v1/beacon/genesis status=200 http_method=GET
DBG 2023-08-25 07:21:26.305+00:00 Received REST response body from remote server remote=127.0.0.1:5052 request=/eth/v1/beacon/genesis contentType=application/json size=169
DBG 2023-08-25 07:21:26.305+00:00 Sending REST request to remote server      remote=127.0.0.1:5052 request=/eth/v1/beacon/states/head/fork http_method=GET
DBG 2023-08-25 07:21:26.306+00:00 Got REST response headers from remote server remote=127.0.0.1:5052 request=/eth/v1/beacon/states/head/fork status=200 http_method=GET
DBG 2023-08-25 07:21:26.306+00:00 Received REST response body from remote server remote=127.0.0.1:5052 request=/eth/v1/beacon/states/head/fork contentType=application/json size=119
DBG 2023-08-25 07:21:26.306+00:00 Sending REST request to remote server      remote=127.0.0.1:5052 request=/eth/v1/config/spec http_method=GET
DBG 2023-08-25 07:21:26.306+00:00 Got REST response headers from remote server remote=127.0.0.1:5052 request=/eth/v1/config/spec status=200 http_method=GET
DBG 2023-08-25 07:21:26.306+00:00 Received REST response body from remote server remote=127.0.0.1:5052 request=/eth/v1/config/spec contentType=application/json size=3859
DBG 2023-08-25 07:21:26.306+00:00 Failed to deserialize REST JSON data       err="<data>(1, 2889) Non-decimal character encountered: 0x00" data="{\"data\":{\"MAX_COMMITTEES_PER_SLOT\":\"64\",\"TARGET_COMMITTEE_SIZE\":\"128\",\"MAX_VALIDATORS_PER_COMMITTEE\":\"2048\",\"SHUFFLE_ROUND_COUNT\":\"90\",\"HYSTERESIS_QUOTIENT\":\"4\",\"HYSTERESIS_DOWNWARD_MULTIPLIER\":\"1\",\"HYSTERESIS_UPWARD_MULTIPLIER\":\"5\",\"MIN_DEPOSIT_AMOUNT\":\"1000000000\",\"MAX_EFFECTIVE_BALANCE\":\"32000000000\",\"EFFECTIVE_BALANCE_INCREMENT\":\"1000000000\",\"MIN_ATTESTATION_INCLUSION_DELAY\":\"1\",\"SLOTS_PER_EPOCH\":\"32\",\"MIN_SEED_LOOKAHEAD\":\"1\",\"MAX_SEED_LOOKAHEAD\":\"4\",\"EPOCHS_PER_ETH1_VOTING_PERIOD\":\"64\",\"SLOTS_PER_HISTORICAL_ROOT\":\"8192\",\"MIN_EPOCHS_TO_INACTIVITY_PENALTY\":\"4\",\"EPOCHS_PER_HISTORICAL_VECTOR\":\"65536\",\"EPOCHS_PER_SLASHINGS_VECTOR\":\"8192\",\"HISTORICAL_ROOTS_LIMIT\":\"16777216\",\"VALIDATOR_REGISTRY_LIMIT\":\"1099511627776\",\"BASE_REWARD_FACTOR\":\"64\",\"WHISTLEBLOWER_REWARD_QUOTIENT\":\"512\",\"PROPOSER_REWARD_QUOTIENT\":\"8\",\"INACTIVITY_PENALTY_QUOTIENT\":\"67108864\",\"MIN_SLASHING_PENALTY_QUOTIENT\":\"128\",\"PROPORTIONAL_SLASHING_MULTIPLIER\":\"1\",\"MAX_PROPOSER_SLASHINGS\":\"16\",\"MAX_ATTESTER_SLASHINGS\":\"2\",\"MAX_ATTESTATIONS\":\"128\",\"MAX_DEPOSITS\":\"16\",\"MAX_VOLUNTARY_EXITS\":\"16\",\"INACTIVITY_PENALTY_QUOTIENT_ALTAIR\":\"50331648\",\"MIN_SLASHING_PENALTY_QUOTIENT_ALTAIR\":\"64\",\"PROPORTIONAL_SLASHING_MULTIPLIER_ALTAIR\":\"2\",\"SYNC_COMMITTEE_SIZE\":\"512\",\"EPOCHS_PER_SYNC_COMMITTEE_PERIOD\":\"256\",\"MIN_SYNC_COMMITTEE_PARTICIPANTS\":\"1\",\"UPDATE_TIMEOUT\":\"8192\",\"INACTIVITY_PENALTY_QUOTIENT_BELLATRIX\":\"16777216\",\"MIN_SLASHING_PENALTY_QUOTIENT_BELLATRIX\":\"32\",\"PROPORTIONAL_SLASHING_MULTIPLIER_BELLATRIX\":\"3\",\"MAX_BYTES_PER_TRANSACTION\":\"1073741824\",\"MAX_TRANSACTIONS_PER_PAYLOAD\":\"1048576\",\"BYTES_PER_LOGS_BLOOM\":\"256\",\"MAX_EXTRA_DATA_BYTES\":\"32\",\"MAX_BLS_TO_EXECUTION_CHANGES\":\"16\",\"MAX_WITHDRAWALS_PER_PAYLOAD\":\"16\",\"MAX_VALIDATORS_PER_WITHDRAWALS_SWEEP\":\"16384\",\"PRESET_BASE\":\"mainnet\",\"CONFIG_NAME\":\"prater\",\"TERMINAL_TOTAL_DIFFICULTY\":\"10790000\",\"TERMINAL_BLOCK_HASH\":\"0x0000000000000000000000000000000000000000000000000000000000000000\",\"TERMINAL_BLOCK_HASH_ACTIVATION_EPOCH\":\"18446744073709551615\",\"MIN_GENESIS_ACTIVE_VALIDATOR_COUNT\":\"16384\",\"MIN_GENESIS_TIME\":\"1614588812\",\"GENESIS_FORK_VERSION\":\"0x00001020\",\"GENESIS_DELAY\":\"1919188\",\"ALTAIR_FORK_VERSION\":\"0x01001020\",\"ALTAIR_FORK_EPOCH\":\"36660\",\"BELLATRIX_FORK_VERSION\":\"0x02001020\",\"BELLATRIX_FORK_EPOCH\":\"112260\",\"CAPELLA_FORK_VERSION\":\"0x03001020\",\"CAPELLA_FORK_EPOCH\":\"162304\",\"DENEB_FORK_VERSION\":\"0x04001020\",\"DENEB_FORK_EPOCH\":\"18446744073709551615\",\"SECONDS_PER_SLOT\":\"12\",\"SECONDS_PER_ETH1_BLOCK\":\"14\",\"MIN_VALIDATOR_WITHDRAWABILITY_DELAY\":\"256\",\"SHARD_COMMITTEE_PERIOD\":\"256\",\"ETH1_FOLLOW_DISTANCE\":\"2048\",\"INACTIVITY_SCORE_BIAS\":\"4\",\"INACTIVITY_SCORE_RECOVERY_RATE\":\"16\",\"EJECTION_BALANCE\":\"16000000000\",\"MIN_PER_EPOCH_CHURN_LIMIT\":\"4\",\"CHURN_LIMIT_QUOTIENT\":\"65536\",\"PROPOSER_SCORE_BOOST\":\"40\",\"DEPOSIT_CHAIN_ID\":\"5\",\"DEPOSIT_NETWORK_ID\":\"5\",\"DEPOSIT_CONTRACT_ADDRESS\":\"0xff50ed3d0ec03ac01d4c79aad74928bff48a7b2b\",\"BLS_WITHDRAWAL_PREFIX\":\"0x00\",\"ETH1_ADDRESS_WITHDRAWAL_PREFIX\":\"0x01\",\"DOMAIN_BEACON_PROPOSER\":\"0x00000000\",\"DOMAIN_BEACON_ATTESTER\":\"0x01000000\",\"DOMAIN_RANDAO\":\"0x02000000\",\"DOMAIN_DEPOSIT\":\"0x03000000\",\"DOMAIN_VOLUNTARY_EXIT\":\"0x04000000\",\"DOMAIN_SELECTION_PROOF\":\"0x05000000\",\"DOMAIN_AGGREGATE_AND_PROOF\":\"0x06000000\",\"TIMELY_SOURCE_FLAG_INDEX\":\"0x00\",\"TIMELY_TARGET_FLAG_INDEX\":\"0x01\",\"TIMELY_HEAD_FLAG_INDEX\":\"0x02\",\"TIMELY_SOURCE_WEIGHT\":\"14\",\"TIMELY_TARGET_WEIGHT\":\"26\",\"TIMELY_HEAD_WEIGHT\":\"14\",\"SYNC_REWARD_WEIGHT\":\"2\",\"PROPOSER_WEIGHT\":\"8\",\"WEIGHT_DENOMINATOR\":\"64\",\"DOMAIN_SYNC_COMMITTEE\":\"0x07000000\",\"DOMAIN_SYNC_COMMITTEE_SELECTION_PROOF\":\"0x08000000\",\"DOMAIN_CONTRIBUTION_AND_PROOF\":\"0x09000000\",\"DOMAIN_BLS_TO_EXECUTION_CHANGE\":\"0x0a000000\",\"TARGET_AGGREGATORS_PER_COMMITTEE\":\"16\",\"RANDOM_SUBNETS_PER_VALIDATOR\":\"1\",\"EPOCHS_PER_RANDOM_SUBNET_SUBSCRIPTION\":\"256\",\"ATTESTATION_SUBNET_COUNT\":\"64\",\"TARGET_AGGREGATORS_PER_SYNC_SUBCOMMITTEE\":\"16\",\"SYNC_COMMITTEE_SUBNET_COUNT\":\"4\"}}"
FAT 2023-08-25 07:21:26.306+00:00 Failed to obtain the config spec of the beacon node reason="Unable to decode REST response, error [Serialization error]"
```

**Additional context**
Nimbus version:
```
Nimbus beacon node v23.8.0-d014d0-stateofus
Copyright (c) 2019-2023 Status Research & Development GmbH

eth2 specification v1.4.0-beta.1

Nim Compiler Version 1.6.14 [Linux: amd64]
```

**Comment by zah** [comment]
We've merged one of the proposed fixes and it would be included in our next release (version 23.9.0, expected in early September).

**Comment by tersec** [comment]
Fixed by https://github.com/status-im/nimbus-eth2/pull/5370 and https://github.com/status-im/nimbus-eth2/pull/5371
